lib/rubocop/cop/style/hash_except.rb

```diff
@@ -187,12 +175,11 @@ def decorate_source(value)
         end
 
         def except_key(node)
-          key_argument = node.argument_list.first.source
-          body = extract_body_if_negated(node.body)
+          key_arg = node.argument_list.first.source
+          body, = extract_body_if_negated(node.body)
           lhs, _method_name, rhs = *body
-          return if [lhs, rhs].map(&:source).none?(key_argument)
 
-          [lhs, rhs].find { |operand| operand.source != key_argument }
+          lhs.source == key_arg ? rhs : lhs
```

Comment:
Since the node pattern now ensures that key is used in the block body, it must be either the receiver or the argument.